app/qml/PositionMarker.qml

```diff
@@ -11,7 +12,62 @@ Item {
     property QgsQuick.PositionKit positionKit
     property color baseColor: InputStyle.highlightColor
     property bool withAccuracy: true
+    property int interval: 200 // Interval of direction marker updates in ms
+    property real threshold: 3 // threshold used to minimized direction marker updates (otherwise shaking)
+    property real speedLimit: 15  // Over speed limit, directions depends on direction of movement
+    property real direction: -1
+    property real groundSpeed: -1
 
+    Timer {
+      interval: positionMarker.interval; running: true; repeat: true
+      onTriggered: {
+        positionMarker.groundSpeed = __inputUtils.groundSpeedFromSource(positionKit)
+        if (groundSpeed > 0) {
+          var newDirection = compass.reading && groundSpeed < speedLimit ? compass.reading.azimuth : positionKit.direction
+        } else {
+          newDirection = compass.reading ? compass.reading.azimuth  : -1
+        }
+
+        var deltaAbsDirection = Math.abs(positionMarker.direction - newDirection)
+        if (deltaAbsDirection > positionMarker.threshold) {
+          positionMarker.direction = newDirection + compass.userOrientation
+        }
+       }
+    }
+
+    OrientationSensor {
+          id: orientationSensor
+          active: true
+
+          Component.onCompleted: {
+            orientationSensor.start()
+          }
+
+          onReadingChanged: {
+            if (reading.orientation == OrientationReading.TopUp) {
+                compass.userOrientation = 0;
+            } else if (reading.orientation == OrientationReading.TopDown) {
+                compass.userOrientation = 180;
+            } else if (reading.orientation == OrientationReading.RightUp) {
+                compass.userOrientation = 90;
+            } else if (reading.orientation == OrientationReading.LeftUp) {
+                compass.userOrientation = 270;
+            }
+          }
+      }
+
+    Compass {
+      id: compass
+      active: true
+      axesOrientationMode: Sensor.AutomaticOrientation
+      skipDuplicates: true
+      bufferSize: 1
```

Comment:
Documentation says "By default, the buffer size is 1, which means no buffering." -- do we need to set this?